src/interchange/avro.rs

```diff
@@ -528,39 +673,26 @@ impl Decoder {
             bail!("wrong avro serialization magic: expected 0, got {}", magic);
         }
 
-        let (resolved_schema, reader_schema) = match &mut self.writer_schemas {
+        let resolved_schema = match &mut self.writer_schemas {
             Some(cache) => match cache.get(schema_id, &self.reader_schema).await? {
                 // If we get a schema back, the writer schema differs from our
                 // schema, so we need to perform schema resolution. If not,
                 // the schemas are identical, so we can skip schema resolution.
-                Some(writer_schema) => (writer_schema, Some(&self.reader_schema)),
-                None => (&self.reader_schema, None),
+                Some(writer_schema) => writer_schema,
+                None => &self.reader_schema,
```

Comment:
The comment above seems stale? Also this is now just `unwrap_or(&self.reader_schema)`, if you like.